server/src/main/java/org/apache/druid/indexing/overlord/supervisor/SupervisorSpec.java

```diff
@@ -71,6 +74,25 @@ default boolean isSuspended()
    */
   String getType();
 
+  /**
+   * @return The types of {@link org.apache.druid.data.input.InputSource} that the task uses. Empty set is returned if
+   * the task does not use any. Users can be given permission to access particular types of
+   * input sources but not others, using the
+   * {@link org.apache.druid.server.security.AuthConfig#enableInputSourceSecurity} config.
+   */
+  default Set<String> getInputSourceTypes() {
```

Comment:
Similar to the comment on `InputSource`: it's better for this to be `Set<Resource>`, so it's clear it's security-related. Method name would be `getInputSourceResources()` in that case.

Reply:
Good suggestion, fixed.